undertow/src/main/java/org/wildfly/extension/undertow/deployment/JBossWebParsingDeploymentProcessor.java

```diff
@@ -74,6 +74,11 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                 warMetaData.setJBossWebMetaData(jBossWebMetaData);
                 // if the jboss-web.xml has a distinct-name configured, then attach the value to this
                 // deployment unit
+                if(jBossWebMetaData.getValves() != null) {
+                    jBossWebMetaData.getValves().stream().forEach((valve) -> {
```

Comment:
this is a real overkill for iterating a list :)
for (Valve valve: jBossWebMetaData.getValves()) would do just fine :)